presto-main/src/main/java/com/facebook/presto/memory/TotalReservationOnBlockedNodesLowMemoryKiller.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.memory;
+
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.memory.MemoryPoolInfo;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.memory.LocalMemoryManager.GENERAL_POOL;
+
+public class TotalReservationOnBlockedNodesLowMemoryKiller
+        implements LowMemoryKiller
+{
+    @Override
+    public Optional<QueryId> chooseQueryToKill(List<QueryMemoryInfo> queries, List<MemoryInfo> nodes)
+    {
+        Map<QueryId, Long> memoryReservationOnBlockedNodes = new HashMap<>();
+        for (MemoryInfo node : nodes) {
+            MemoryPoolInfo generalPool = node.getPools().get(GENERAL_POOL);
+            if (generalPool == null) {
+                continue;
+            }
+            if (generalPool.getFreeBytes() + generalPool.getReservedRevocableBytes() > 0) {
+                continue;
+            }
+            Map<QueryId, Long> queryMemoryReservations = generalPool.getQueryMemoryReservations();
+            for (Map.Entry<QueryId, Long> query : queryMemoryReservations.entrySet()) {
+                QueryId queryId = query.getKey();
+                long memoryReservation = query.getValue();
+                memoryReservationOnBlockedNodes.compute(queryId, (id, oldValue) -> oldValue == null ? memoryReservation : oldValue + memoryReservation);
+            }
+        }
+
+        QueryId biggestQuery = null;
+        long maxMemory = 0;
+        for (Map.Entry<QueryId, Long> query : memoryReservationOnBlockedNodes.entrySet()) {
```

Comment:
I believe there is a stream form for this... stream the entries, and get max by a custom ordering of entry.getValue

Reply:
You seem to be referring to `Collectors.maxBy`